src/NSwag.Core/OpenApiParameter.cs

```diff
@@ -44,6 +44,10 @@ public string Name
             }
         }
 
+        /// <summary>Gets or sets a custom name which is often used in code generation (default: null).</summary>
+        [JsonProperty(PropertyName = "x-name", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
```

Comment:
Better use x-parameterName to avoid collisions with other tools.

Reply:
`x-parameter-name` or `x-original-parameter` would be better names IMHO

Reply:
All properties in JSON Schema/OpenAPI use lower camel case, e.g. "securityDefinitions" - so i would go with x-lowerCamelCase, no?

I like x-originalParameter better.

Reply:
Then ok for `x-originalParameter`